lib/Yam/Installation/Upgrade/SelectUpdatePage.pm

```diff
@@ -0,0 +1,32 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Handle page that allows to select system/partition to be upgraded.
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+package Yam::Installation::Upgrade::SelectUpdatePage;
+use parent 'Installation::Navigation::NavigationBase';
+use strict;
+use warnings;
+
+sub init {
+    my ($self) = @_;
+    $self->SUPER::init();
+    $self->{tbl_select_origin} = $self->{app}->table({id => "partition"});
+    return $self;
+}
+
+sub get_select_origin {
```

Comment:
now we have all the logic in one page, without controllers, what we are doing is to preceed with `_` to distinguish private method that public interface and we also will put all the public method first.
So this one can be move down and be called `_get_select_for_update_page` (which will never be called from outside).